File: musifan/src/main/java/musifan/musifan/dto/Artiste.java
package musifan.musifan.dto;

import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonView;

import musifan.musifan.entity.JsonViews;

public class Artiste {
	@JsonView({JsonViews.Common.class,})
	private Long id;
	@JsonView({JsonViews.Common.class,})
	private String mail;
	private String password;
	@JsonView({JsonViews.Common.class,})
	private String nom;
	@JsonView({JsonViews.Common.class,})
	private String prenom;
	@JsonView({JsonViews.Common.class,})
	private String telephone;
	@JsonView({JsonViews.Common.class,})
	private byte[] photoProfil;
	@JsonView({JsonViews.Common.class,})
	private byte[] photoBanniere;
	@JsonView({JsonViews.Common.class,})
	private String nomArtiste;
	@JsonView({JsonViews.Common.class,})
	private String description;
	@JsonView({JsonViews.ArtisteComplet.class, JsonViews.UtilisateurAvecAlbumsArtiste.class})
	private Set<Album> albums = new HashSet<Album>();
	@JsonView({JsonViews.ArtisteComplet.class,})
	private Set<Concert> concerts = new HashSet<Concert>();
	@JsonView({JsonViews.ArtisteComplet.class,JsonViews.UtilisateurAvecPublicationsArtiste.class})
	private Set<Publication> publications = new HashSet<Publication>();
	
	public Artiste() {
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getMail() {
		return mail;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public byte[] getPhotoProfil() {
		return photoProfil;
	}

	public void setPhotoProfil(byte[] photoProfil) {
		this.photoProfil = photoProfil;
	}

	public byte[] getPhotoBanniere() {
		return photoBanniere;
	}

	public void setPhotoBanniere(byte[] photoBanniere) {
		this.photoBanniere = photoBanniere;
	}

	public String getNomArtiste() {
		return nomArtiste;
	}

	public void setNomArtiste(String nomArtiste) {
		this.nomArtiste = nomArtiste;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Set<Album> getAlbums() {
		return albums;
	}

	public void setAlbums(Set<Album> albums) {
		this.albums = albums;
	}

	public Set<Concert> getConcerts() {
		return concerts;
	}

	public void setConcerts(Set<Concert> concerts) {
		this.concerts = concerts;
	}

	public Set<Publication> getPublications() {
		return publications;
	}

	public void setPublications(Set<Publication> publications) {
		this.publications = publications;
	}
	
	
	
	
}

File: musifanAngular/src/app/models/artiste.ts
import { Album } from "./album";
import { Publication } from "./publication";
import { Concert } from "./concert";
import { Byte } from "@angular/compiler/src/util";

export class Artiste {
  public constructor(
    private _id?: number | undefined,
    private _mail?: string | undefined,
    private _nom?: string | undefined,
    private _prenom?: string | undefined,
    private _nomArtiste?: string | undefined,
    private _telephone?: string | undefined,
    private _description?: string | undefined,
    private _publications?: Publication[] | undefined,
    private _albums?: Album[] | undefined,
    private _concerts?: Concert[] | undefined,
    private _photoProfil?: Byte[] | undefined,
    private _photoBanniere?: Byte[] | undefined
  ) {}

  /**
   * Getter id
   * @return {number | undefined}
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter mail
   * @return {string | undefined}
   */
  public get mail(): string | undefined {
    return this._mail;
  }

  /**
   * Getter nom
   * @return {string | undefined}
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Getter prenom
   * @return {string | undefined}
   */
  public get prenom(): string | undefined {
    return this._prenom;
  }

  /**
   * Getter nomArtiste
   * @return {string | undefined}
   */
  public get nomArtiste(): string | undefined {
    return this._nomArtiste;
  }

  /**
   * Getter telephone
   * @return {string | undefined}
   */
  public get telephone(): string | undefined {
    return this._telephone;
  }

  /**
   * Getter description
   * @return {string | undefined}
   */
  public get description(): string | undefined {
    return this._description;
  }

  /**
   * Getter publications
   * @return {Publication[] | undefined}
   */
  public get publications(): Publication[] | undefined {
    return this._publications;
  }

  /**
   * Getter albums
   * @return {Album[] | undefined}
   */
  public get albums(): Album[] | undefined {
    return this._albums;
  }

  /**
   * Getter concerts
   * @return {Concert[] | undefined}
   */
  public get concerts(): Concert[] | undefined {
    return this._concerts;
  }

  /**
   * Getter photoProfil
   * @return {Byte[] | undefined}
   */
  public get photoProfil(): Byte[] | undefined {
    return this._photoProfil;
  }

  /**
   * Getter photoBanniere
   * @return {Byte[] | undefined}
   */
  public get photoBanniere(): Byte[] | undefined {
    return this._photoBanniere;
  }

  /**
   * Setter id
   * @param {number | undefined} value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter mail
   * @param {string | undefined} value
   */
  public set mail(value: string | undefined) {
    this._mail = value;
  }

  /**
   * Setter nom
   * @param {string | undefined} value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Setter prenom
   * @param {string | undefined} value
   */
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }

  /**
   * Setter nomArtiste
   * @param {string | undefined} value
   */
  public set nomArtiste(value: string | undefined) {
    this._nomArtiste = value;
  }

  /**
   * Setter telephone
   * @param {string | undefined} value
   */
  public set telephone(value: string | undefined) {
    this._telephone = value;
  }

  /**
   * Setter description
   * @param {string | undefined} value
   */
  public set description(value: string | undefined) {
    this._description = value;
  }

  /**
   * Setter publications
   * @param {Publication[] | undefined} value
   */
  public set publications(value: Publication[] | undefined) {
    this._publications = value;
  }

  /**
   * Setter albums
   * @param {Album[] | undefined} value
   */
  public set albums(value: Album[] | undefined) {
    this._albums = value;
  }

  /**
   * Setter concerts
   * @param {Concert[] | undefined} value
   */
  public set concerts(value: Concert[] | undefined) {
    this._concerts = value;
  }

  /**
   * Setter photoProfil
   * @param {Byte[] | undefined} value
   */
  public set photoProfil(value: Byte[] | undefined) {
    this._photoProfil = value;
  }

  /**
   * Setter photoBanniere
   * @param {Byte[] | undefined} value
   */
  public set photoBanniere(value: Byte[] | undefined) {
    this._photoBanniere = value;
  }
}
